CMakeLists.txt

```
@@ -0,0 +1,5 @@
+cmake_minimum_required(VERSION 2.6)
```

You need at least 3.8 to support
```
set(CMAKE_CXX_STANDARD 17)
```
Which you call a few lines lower.
```suggestion
cmake_minimum_required(VERSION 3.8)
```